### DISCUSSION TITLE
LINQ -  Where Query sample - Degenerate query

### DISCUSSION THREAD

**Discussion by bernd5** [dotnet/csharplang#5436] [General]
The [spec ](https://github.com/dotnet/csharplang/blob/main/spec/expressions.md#select-clauses) says in a sample that:

```csharp
from c in customers.Where(c => c.City == "London")
select c
```

is translated to:

```csharp
customers.Where(c => c.City == "London")
```

which is not what actually happens because (e.g. roslyn) does not interpret the Where method call as part of the Linq query and produces:

```csharp
customers.Where(c => c.City == "London").Select(c => c);
````

For roslyn it is a `degenerate query` (see [link](https://github.com/dotnet/csharplang/blob/main/spec/expressions.md#degenerate-query-expressions)).

Shouldn't the sample changed to:

```csharp
from c in customers
where c.City == "London"
select c
```

**Comment by 333fred** [comment]
This just looks like a roslyn bug: the example is saying that we should be translating this code into a degenerate query, but we are not doing so.

**Reply by yaakov-h** [reply]
When compared to the sections above and below it, though, this particular example seems inconsistent, and the OP’s suggestion would be far more at home in that document.
